Packs/Infinipoint/Integrations/Infinipoint/Infinipoint.py

```diff
@@ -0,0 +1,465 @@
+    if last_fetch is None:
+        last_fetch = first_fetch_time
+    else:
+        last_fetch = int(last_fetch)
+
+    latest_created_time = cast(int, last_fetch)
+    incidents: List[Dict[str, Any]] = []
+
+    args = {
+        'limit': max_results,
+        'offset': last_fetch
+    }
+
+    alerts = infinipoint_command(client, args, COMMANDS_CONFIG['infinipoint-get-events'])
+
+    if alerts:
+        for alert in alerts.outputs:
+            if alert.get("subscription") in subscription:
+                incident_created_time = int(alert.get('timestamp', '0'))
+                incident_created_time_ms = incident_created_time * 1000
+
+                incident = {
+                    'name': f'Infinipoint {alert.get("name")}',
+                    'type': f'Infinipoint {alert.get("type")}',
+                    'occurred': timestamp_to_datestring(incident_created_time_ms),
+                    'rawJSON': json.dumps(alert.get('rawJSON'))
+                }
+
+                incidents.append(incident)
+                if incident_created_time > latest_created_time:
+                    latest_created_time = incident_created_time
+
+    next_run = {'last_fetch': latest_created_time}
+
+    demisto.setLastRun(next_run)
+    demisto.incidents(incidents)
+
+
+'''MAIN FUNCTIONS'''
+
+
+def test_module(route, base_url, insecure, headers):
+    """Tests API connectivity and authentication'
+    Returning '200' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    """
+    res = requests.request(
+        "POST",
+        base_url + route,
+        headers=headers,
+        verify=insecure
+    )
+    res.raise_for_status()
+
+
+def infinipoint_command(client: Client, args=None, optional_args=None, pagination=True):
+    rules = None
+    cve = []
+    method = "POST"
+
+    # Cancel pagination if necessary
+    if "pagination" in optional_args:
+        pagination = optional_args['pagination']
+
+    # Pass arguments as is
+    if "pass_args" in optional_args:
+        rules = args
+    # Move request type to GET
+    elif "get_req" in optional_args:
+        optional_args['route'] = optional_args['route'].format(**args)
+        method = "GET"
+
+    # Change url - Post request
+    elif "format_route" in optional_args:
+        optional_args['route'] = optional_args['route'].format(**args)
+
+    else:
+        rules = []
+        for k, v in optional_args['args'].items():
+            if args.get(k):
+                rules.append({'field': k, "operator": v, "value": f"{args[k]}"})
+
+    res = client.call_api(optional_args['route'], rules, pagination=pagination, method=method)
+
+    if res:
+        for node in res:
+            # Handle time format - convert to ISO from epoch
+            if '$time' in node and isinstance(node['$time'], int):
+                created_time_ms = int(node.get('$time', '0')) * 1000
+                node['$time'] = timestamp_to_datestring(created_time_ms)
```

Comment:
```suggestion
                created_time = datetime.fromtimestamp(int(node.get('$time', '0')), timezone.utc)
                node['$time'] = created_time.isoformat()
```
Don't forget to add  `from datetime import timezone`